src/v1/destinations/algolia/networkHandler.js

```diff
@@ -0,0 +1,84 @@
+/* eslint-disable no-restricted-syntax */
+const { TransformerProxyError } = require('../../../v0/util/errorTypes');
+const { prepareProxyRequest, proxyRequest } = require('../../../adapters/network');
+const { isHttpStatusSuccess, getAuthErrCategoryFromStCode } = require('../../../v0/util/index');
+
+const {
+  processAxiosResponse,
+  getDynamicErrorType,
+} = require('../../../adapters/utils/networkUtils');
+const tags = require('../../../v0/util/tags');
+
+const responseHandler = (responseParams) => {
+  const { destinationResponse, rudderJobMetadata } = responseParams;
+  const message = `[ALGOLIA Response V1 Handler] - Request Processed Successfully`;
+  const responseWithIndividualEvents = [];
+  // response:
+  // {status: 200, message: 'OK'}
+  // {response:'[ENOTFOUND] :: DNS lookup failed', status: 400}
+  // destinationResponse = {
+  //   response: {"status": 422, "message": "EventType must be one of \"click\", \"conversion\" or \"view\""}, status: 422
+  // }
+  const { response, status } = destinationResponse;
+
+  if (isHttpStatusSuccess(status)) {
+    for (const mData of rudderJobMetadata) {
+      const proxyOutputObj = {
+        statusCode: 200,
+        metadata: mData,
+        error: 'success',
+      };
+      responseWithIndividualEvents.push(proxyOutputObj);
+    }
+
+    return {
+      status,
+      message,
+      destinationResponse,
+      response: responseWithIndividualEvents,
+    };
+  }
+
+  // in case of non 2xx status sending 500 for every event, populate response and update dontBatch to true
```

Comment:
This statement is not correct we are returning 400 when `dontBatch` is true and status is 422.

Reply:
we sending 400 as 422 won't be handled in server, 500 will be sent first time

Reply:
Can you update the statement then accordingly?